chain/jsonrpc/src/lib.rs

```diff
@@ -163,24 +179,43 @@ impl JsonRpcHandler {
     async fn send_tx_commit(&self, params: Option<Value>) -> Result<Value, RpcError> {
         let tx = parse_tx(params)?;
         let tx_hash = tx.get_hash();
+        let signer_account_id = tx.transaction.signer_id.clone();
         let result = self
             .client_addr
             .send(NetworkClientMessages::Transaction(tx))
             .map_err(|err| RpcError::server_error(Some(err.to_string())))
             .compat()
             .await?;
         match result {
-            NetworkClientResponses::ValidTx => {
+            NetworkClientResponses::ValidTx | NetworkClientResponses::NoResponse => {
```

Comment:
`NoResponse` should not mean "needs routing", we should add `NeedsRouting` response.